docs/src/content/docs/cookbook/dexes/dedust.mdx

```diff
@@ -7,8 +7,430 @@ sidebar:
+    // Defaults to `null`, which makes the swap use the sender's address
+    recipientAddress: Address? = null;
+
+    // Referral address, required for the referral program of DeDust
+    // Defaults to `null`
+    referralAddress: Address? = null;
+
+    // Custom payload that will be attached to the fund transfer upon a successful swap
+    // Defaults to `null`
+    fulfillPayload: Cell? = null;
+
+    // Custom payload that will be attached to the fund transfer upon a rejected swap
+    // Defaults to `null`
+    rejectPayload: Cell? = null;
+}
+```
+
+### Swap Toncoin for any Jetton
+
+```tact
+/// https://docs.dedust.io/reference/tlb-schemes#message-swap
+message(0xea06185d) NativeSwap {
+    // Unique identifier used to trace transactions across multiple contracts and to prevent replay attacks
+    // Defaults to 0, which means Native Vault will handle this for us
+    queryId: Int as uint64 = 0;
+
+    // Toncoin amount for the swap
+    amount: Int as coins;
+
+    // Inlined fields of SwapStep Struct
+    poolAddress: Address;
+    kind: Int as uint1 = 0;
+    limit: Int as coins = 0;
+    nextStep: SwapStep? = null;
+
+    // Set of parameters relevant for the whole swap
+    swapParams: SwapParams;
+}
+
+// Let's say `swapAmount` is `ton("0.1")`, which is 10000000 nanoToncoins
+fun swapToncoinForUSDT(swapAmount: Int) {
+    send(SendParameters{
+        // Address of TON vault to send the message to
+        to: address("EQDa4VOnTYlLvDJ0gZjNYm5PXfSmmtL6Vs6A_CZEtXCNICq_"),
+        // Amount to swap plus a trade fee
+        value: swapAmount + ton("0.2"),
+        body: NativeSwap{
+            amount: swapAmount,
+            // Address of the swap pool, which is the TON/USDT pair in this case
+            poolAddress: address("EQCxE6mUtQJKFnGfaROTKOt1lZbDiiX1kCixRv7Nw2Id_sDs"),
+            // Set of parameters relevant for the whole swap
+            swapParams: SwapParams{}, // use defaults
+        }.toCell(),
+    });
+}
+
+//
+// Helper Structs described earlier on this page
+//
+
+struct SwapStep {
+    poolAddress: Address;
+    kind: Int as uint1 = 0;
+    limit: Int as coins = 0;
+    nextStep: Cell?;
+}
+
+struct SwapParams {
+    deadline: Int as uint32 = 0;
+    recipientAddress: Address? = null;
+    referralAddress: Address? = null;
+    fulfillPayload: Cell? = null;
+    rejectPayload: Cell? = null;
+}
+```
+
+### Swap a Jetton for another Jetton or Toncoin
+
+```tact
+/// https://docs.dedust.io/reference/tlb-schemes#message-swap-1
+message(0xe3a0d482) JettonSwapPayload {
+    // Inlined fields of SwapStep Struct
+    poolAddress: Address;
+    kind: Int as uint1 = 0;
+    limit: Int as coins = 0;
+    nextStep: SwapStep? = null;
+
+    // Set of parameters relevant for the whole swap
+    swapParams: SwapParams;
+}
+
+fun swapJetton() {
+    // NOTE: Calculate wallet address of Jetton for the target user
+    let targetJettonWalletAddress = address("");
```

Comment:
we should explain how or at least provide a link to related docs page

Reply:
Provided links